src/main/java/attendance/domain/attendance/Attendance.java

```diff
@@ -0,0 +1,108 @@
+package attendance.domain.attendance;
+
+import static attendance.common.SystemDateConfig.*;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+
+import attendance.common.exception.AttendanceArgumentException;
+import attendance.domain.AttendanceStatus;
+
+public record Attendance(LocalDateTime dateTime, AttendanceStatus attendanceStatus) {
+
+    public Attendance(LocalDateTime dateTime) {
+        this(dateTime, decideAttendanceStatus(dateTime));
+    }
+
+    private static AttendanceStatus decideAttendanceStatus(LocalDateTime dateTime) {
+        validate(dateTime);
+        LocalTime baseSchedule = getScheduleForDay(dateTime);
+        return decideOnDefaultDay(dateTime.toLocalTime(), baseSchedule);
+    }
+
+    private static void validate(LocalDateTime dateTime) {
+        var date = dateTime.toLocalDate();
+        validateIsWeekend(date);
+        validateIsHoliday(date);
+        validateIsDuringCampusSchedule(dateTime);
+    }
+
+    private static void validateIsHoliday(LocalDate date) {
+        boolean isHoliday = DAT_OF_HOLIDAY.stream()
+            .anyMatch(day -> date.getDayOfMonth() == day);
+
+        if (isHoliday) {
+            throw new AttendanceArgumentException(Message.CANNOT_ATTENDANCE_WEEKEND_FORMAT, date);
+        }
+    }
+
+    private static void validateIsWeekend(LocalDate date) {
+        if (date.getDayOfWeek().getValue() >= DayOfWeek.SATURDAY.getValue()) {
+            throw new AttendanceArgumentException(Message.CANNOT_ATTENDANCE_WEEKEND_FORMAT, date);
+        }
+    }
+
+    private static void validateIsDuringCampusSchedule(LocalDateTime dateTime) {
+        LocalTime time = dateTime.toLocalTime();
+        if (isDuringCampusSchedule(time)) {
+            throw new AttendanceArgumentException(Message.OUT_OF_SCHOOL_SCHEDULE);
+        }
+    }
+
+    private static boolean isDuringCampusSchedule(LocalTime time) {
+        return time.isBefore(Schedule.CAMPUS_OPEN) || time.isAfter(Schedule.CAMPUS_CLOSE);
+    }
```

Comment:
```suggestion
    private static void validateIsDuringCampusSchedule(LocalDateTime dateTime) {
        LocalTime time = dateTime.toLocalTime();
        if (time.isBefore(Schedule.CAMPUS_OPEN) || time.isAfter(Schedule.CAMPUS_CLOSE)) {
            throw new AttendanceArgumentException(Message.OUT_OF_SCHOOL_SCHEDULE);
        }
    }
```
if문 하나만 바라보아도 되는 상황에서 굳이 메소드로 분리할 필요는 없어보여요.